src/functional/ffront/decorator.py

```diff
@@ -231,6 +239,45 @@ def _validate_args(self, *args, **kwargs) -> None:
             raise NotImplementedError("Keyword arguments are not supported yet.")
 
     def __call__(self, *args, offset_provider: dict[str, Dimension], **kwargs) -> None:
+        rewritten_args, size_args, kwargs = self._process_args(args, kwargs)
+
+        backend = self.backend
+        if not self.backend:
+            warnings.warn(
+                UserWarning(
+                    f"Field View Program '{self.itir.id}': Using default ({DEFAULT_BACKEND}) backend."
+                )
+            )
+            backend = DEFAULT_BACKEND
```

Comment:
It looks cleaner to me this way, but it's just a suggestion. Feel free to ignore:
```suggestion
        if not self.backend:
            warnings.warn(
                UserWarning(
                    f"Field View Program '{self.itir.id}': Using default ({DEFAULT_BACKEND}) backend."
                )
            )
            backend = DEFAULT_BACKEND
        else:
            backend = self.backend
```

Reply:
the suggestion seems incomplete to me, it would leave the following state:

```python
backend = self.backend
if not self.backend:
    ...
    backend = DEFAULT_BACKEND
else:
    backend = self.backend  # <-- seems pointless
```

Did you mean the following?

```python
# backend undefined at this point
if self.backend:
    backend = self.backend
else:
    ...
    backend = DEFAULT_BACKEND
```

I didn't do it like that purely out of personal preference.